compiler-core/src/exhaustiveness.rs

```diff
@@ -1314,6 +1270,59 @@ impl BitArrayMatchedValue {
             | BitArrayMatchedValue::Discard(_) => None,
         }
     }
+
+    /// Returns true if the two `MatchedValue`s are matching for the exact same
+    /// thing. Note how this doesn't automatically mean they will be matching
+    /// for the exact same thing in a whole bit array pattern as they could be
+    /// matching at different positions!
+    ///
+    /// ```gleam
+    /// <<1, _>>
+    /// <<_, 1>>
+    /// // Both are matching on the same value but at different positions!
+    /// ```
+    ///
+    fn checking_same_value(&self, other: &Self) -> bool {
```

Comment:
If this is used only in that one test please move it to the test 🙏

Reply:
There it used the Eq implementation but I needed to check for sameness and not strict equality only in the `fails_if_failing` (and the other) functions. In all the other places there's no need to change implementation and it's ok to still use Eq like we were doing before